main-actions/src/main/scala/sbt/Tests.scala

```diff
@@ -39,6 +37,12 @@ sealed trait TestOption
 
 object Tests {
 
+  final case class TestDefinitionsWithFramework(
+      testPerFramework: Map[Framework, Seq[TestDefinition]]
+  ) {
+    def allDefinitions: Seq[TestDefinition] = testPerFramework.values.flatten.toVector
```

Comment:
Utility method because previously there was no grouping by the test framework. For not BSP purpose mapping `framework -> tests` isn't useful but I didn't want to create separate logic just for BSP. What do you think of it?